src/line.rs

```diff
@@ -260,7 +294,18 @@ impl<'input, Endian> StateMachine<'input, Endian>
             }
 
             Opcode::DefineFile(entry) => {
-                self.header.file_names.push(entry);
+                match self.header {
+                    LineNumberProgramHeaderHolder::Incomplete(ref mut header) => {
+                        header.file_names.push(entry);
+                    },
+                    _ => {},
+                    /*
+                    LineNumberProgramHeaderHolder::Complete(_) => {
+                        // All the filenames are already in the header, we don't
+                        // need to do anything (except have this branch to get
+                        // rustc not to warn about an unused 'Complete').
```

Comment:
I like the commented out version because

* If we add more variants (however unlikely) then the compiler will yell at us to match them
* The comment is nice